be/src/storage/task/engine_clone_task.cpp

```diff
@@ -209,6 +209,13 @@ Status EngineCloneTask::_do_clone_primary_tablet(Tablet* tablet) {
     vector<int64_t> missing_version_ranges;
     st = tablet->updates()->get_missing_version_ranges(missing_version_ranges);
     if (st.ok()) {
+        // Probably no missing version in this case, we just return FE the clone success to avoid wasted work.
+        // If there is indeed missing version, FE will schedule clone again.
+        if (missing_version_ranges.size() == 1 && _clone_req.committed_version < missing_version_ranges.back()) {
+            LOG(INFO) << "Cloning existing tablet skipped, no missing version. tablet:" << tablet->tablet_id()
+                      << " type:" << KeysType_Name(tablet->keys_type()) << " version:" << _clone_req.committed_version;
+            return st;
+        }
```

Comment:
The clone should only consider src&dest's version(or this tablet's max version, not partitions commit_version),  for example: 
```
src version: 10
dest version: 8
committed version: 9
```
In the scenario above,  clone will never be invoked properly.

Reply:
I know that, but currently `get_missing_version_ranges()` will always return non-empty result even though there is no missing  version and that's why we see lots of `get_rowsets_for_snapshot: no version to clone tablet...` warning logs. Because although FE thought dest BE missed some versions when checking, but after clone task is sent to BE, BE already published that missing version successfully(as the code explained `missing_version_ranges.size() == 1 && _clone_req.committed_version < missing_version_ranges.back()`, ps: this logic is hard to understand, i should still change the `get_missing_version_ranges()` function) and no version need to be cloned.

There is chance that there are new versions published after the `committed_version`, but i think that chance is rare, so without some kind of negotiation mechanism with src replica, we just ignore that chance.